rules.ts

```diff
@@ -0,0 +1,113 @@
+// Fizz Buzz with rules
+
+interface Rule {
+    match(x: number): boolean;
+    apply(acc: string[]): void;
+}
+
+class RuleFizz implements Rule {
+    match(x: number): boolean {
+        return x % 3 == 0;
+    }
+
+    apply(acc: string[]): void {
+        acc.push("Fizz")
+    }
+}
+
+class RuleFezz implements Rule {
+    match(x: number): boolean {
+        return x % 13 == 0;
+    }
+
+    apply(acc: string[]): void {
+        acc.push("Fezz")
+    }
+}
+
+class RuleBuzz implements Rule {
+    match(x: number): boolean {
+        return x % 5 == 0;
+    }
+
+    apply(acc: string[]): void {
+        acc.push("Buzz")
+    }
+}
+
+class RuleBang implements Rule {
+    match(x: number): boolean {
+        return x % 7 == 0;
+    }
+
+    apply(acc: string[]): void {
+        acc.push("Bang")
+    }
+}
+
+class RuleBong implements Rule {
+    match(x: number): boolean {
+        return x % 11 == 0;
+    }
+
+    apply(acc: string[]): void {
```

Comment:
We could come up with a better name for `acc`

Reply:
Acc comes from accumulator, I will try to think of a better name (maybe cur_res?).